cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -1008,3 +1026,72 @@ func (ht *hcsTask) requestUpdateContainer(ctx context.Context, resourcePath stri
 	}
 	return ht.c.Modify(ctx, modification)
 }
+
+// publishNotifications monitors the containers notification queue.
+// Currently only OOM events are published.
+//
+// This MUST be called via a goroutine to wait on a background thread.
+func (ht *hcsTask) publishNotifications(ctx context.Context) {
```

Comment:
I think I would prefer for this function call to return a named value for the error instead. That would make the logic for setting the span status and the later error values easier to reason about.

Reply:
it doesn't have to be, it can be declared with a `var err error`, which I think scopes an initializes it the same as a return type

Reply:
either way works I suppose. I'm more of a fan of letting the caller decide to ignore errors or not, but the current way looks like it's the pattern that the other calls when making a task use.

Reply:
Yea, my head was in the below case 

```go
go ht.publishNotifications(ctx)
```